src/main/java/com/brightpath/learnify/domain/auth/PermissionAccessService.java

```diff
@@ -0,0 +1,120 @@
+package com.brightpath.learnify.domain.auth;
+
+import com.brightpath.learnify.domain.auth.permission.ResourceAccessEnum;
+import com.brightpath.learnify.domain.auth.permission.Permission;
+import com.brightpath.learnify.domain.auth.permission.PermissionLevel;
+import com.brightpath.learnify.domain.auth.permission.ResourceAccessSummary;
+import com.brightpath.learnify.domain.common.ResourceType;
+import com.brightpath.learnify.domain.common.UuidProvider;
+import com.brightpath.learnify.exception.notfound.ResourceNotFoundException;
+import com.brightpath.learnify.persistance.auth.permissions.PermissionEntity;
+import com.brightpath.learnify.persistance.auth.permissions.PermissionRepository;
+import com.brightpath.learnify.persistance.auth.permissions.PermissionsAccessEntity;
+import com.brightpath.learnify.persistance.auth.permissions.PermissionsAccessRepository;
+import com.brightpath.learnify.persistance.common.PersistentMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class PermissionAccessService {
+
+    private final PermissionsAccessRepository permissionAccessRepository;
+
+    private final PermissionRepository permissionRepository;
+
+    private final PersistentMapper persistentMapper;
+
+    private final UuidProvider uuidProvider;
+
+
+    // method for checking if user has access to a resource with a given id and type
+    public ResourceAccessEnum getUserAccessForResource(String userId, UUID resourceId, ResourceType resourceType) {
+        ResourceAccessSummary access = permissionAccessRepository.findUserAccessToResource(userId, permissionAccessId(resourceId, resourceType));
+        if (access.ownerId().equals(userId)) {
+            return ResourceAccessEnum.OWNER;
+        }
+        return switch (access.permissionLevel()) {
+            case PUBLIC -> {
+                if (access.resourceAccessEnum() == null) {
+                    yield ResourceAccessEnum.READ_WRITE;
+                }else {
+                    yield ResourceAccessEnum.DENIED;
```

Comment:
here should be returned access.resourceAccessEnum() instead of DENIED

Reply:
add static import